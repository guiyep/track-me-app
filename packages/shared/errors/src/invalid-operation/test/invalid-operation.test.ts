import { InvalidOperation } from '../index';

describe('InvalidOperation', () => {
  it('should create an instance with the correct properties', () => {
    const message = 'Test error message';
    const error = new InvalidOperation({ message });

    expect(error).toBeInstanceOf(InvalidOperation);
    expect(error).toBeInstanceOf(Error);
    expect(error).toMatchSnapshot();
    expect(error.errorNumber).toBe(400);
    expect(error.type).toBe('InvalidOperation');
    expect(error.message).toBe(message);
  });

  it('should maintain the error stack trace', () => {
    const error = new InvalidOperation({ message: 'Test' });
    expect(error.stack).toBeDefined();
    expect(typeof error.stack).toBe('string');
  });
});
