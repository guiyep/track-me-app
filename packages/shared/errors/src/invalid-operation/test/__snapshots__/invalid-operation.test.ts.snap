// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidOperation should create an instance with the correct properties 1`] = `[Error: Test error message]`;
